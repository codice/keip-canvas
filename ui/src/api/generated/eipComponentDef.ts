/* eslint-disable */
/**
 * This file was automatically generated by json-schema-to-typescript.
 * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
 * and run json-schema-to-typescript to regenerate this file.
 */

/**
 * A top-level EIP component
 */
export type EipComponent = EipElement & {
  role: EipRole;
  connectionType: ConnectionType;
};
/**
 * The attribute's value type (attribute keys are always strings)
 */
export type AttributeType = string | number | boolean;
/**
 * A child EIP element. Each child element is a descendant of a top-level EipComponent and can recursively contain more child elements
 */
export type EipChildElement = EipElement & {
  occurrence?: Occurrence;
};
/**
 * Describes the expected general behavior of an EIP component
 */
export type EipRole = "channel" | "endpoint" | "router" | "transformer";
/**
 * Defines a connection pattern for an EIP component
 */
export type ConnectionType =
  | "content_based_router"
  | "inbound_request_reply"
  | "passthru"
  | "request_reply"
  | "sink"
  | "source"
  | "tee";

/**
 * Defines the collection of EIP components available for use
 */
export interface EipComponentDefinition {
  [k: string]: EipComponent[];
}
/**
 * The base schema for all EIP Elements (e.g. EipComponent, EipChildElement)
 */
export interface EipElement {
  name: string;
  description?: string;
  attributes?: Attribute[];
  childGroup?: EipChildGroup;
}
/**
 * Defines an EIP attribute
 */
export interface Attribute {
  name: string;
  type: "string" | "boolean" | "number";
  description?: string;
  default?: AttributeType;
  required?: boolean;
  restriction?: Restriction;
}
/**
 * Configures a constraint on the possible values of an attribute. Currently, only string enum constraints are supported.
 */
export interface Restriction {
  type: "enum";
  values?: string[];
}
/**
 * A container for child elements used to indicate their order or mutual exclusivity
 */
export interface EipChildGroup {
  /**
   * The type of child grouping. When combined with each child element's 'Occurrence', constrains the child's order and cardinality
   */
  indicator: "all" | "choice" | "sequence";
  children: EipChildElement[];
  occurrence?: Occurrence;
}
/**
 * Determines the cardinality of each element in a group
 */
export interface Occurrence {
  /**
   * The minimum number of instances of this element in the group
   */
  min?: number;
  /**
   * The maximum number of instances of this element in the group. Setting this to a value of '-1' enables adding an unbounded number of elements
   */
  max?: number;
}
