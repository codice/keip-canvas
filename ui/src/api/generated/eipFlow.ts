/* eslint-disable */
/**
 * This file was automatically generated by json-schema-to-typescript.
 * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
 * and run json-schema-to-typescript to regenerate this file.
 */

/**
 * Describes the expected general behavior of an EIP component
 */
export type EipRole = "channel" | "endpoint" | "router" | "transformer";
/**
 * Defines a connection pattern for an EIP component
 */
export type ConnectionType =
  | "content_based_router"
  | "inbound_request_reply"
  | "passthru"
  | "request_reply"
  | "sink"
  | "source"
  | "tee";
/**
 * The attribute's value type (attribute keys are always strings)
 */
export type AttributeType = string | number | boolean;

/**
 * A representation of an EIP flow diagram as a collection of nodes and their corresponding edges
 */
export interface EipFlow {
  nodes?: EipNode[];
  edges?: FlowEdge[];
}
/**
 * An instance of an 'EipComponent' as a node in the flow diagram
 */
export interface EipNode {
  id: string;
  eipId: EipId;
  label?: string;
  description?: string;
  role: EipRole;
  connectionType: ConnectionType;
  attributes?: Attributes;
  children?: EipChildNode[];
}
/**
 * A combination of fields uniquely identifying a component in an 'EipComponentDefinition'
 */
export interface EipId {
  namespace: string;
  name: string;
}
/**
 * A map of EIP attributes belonging to the node
 */
export interface Attributes {
  [k: string]: AttributeType;
}
/**
 * An instance of an 'EipChildElement'. Can recursively contain more child nodes as well as attributes of its own.
 */
export interface EipChildNode {
  name: string;
  attributes?: Attributes;
  children?: EipChildNode[];
}
/**
 * A single edge in the flow diagram connecting two nodes
 */
export interface FlowEdge {
  id: string;
  source: string;
  target: string;
  type?: "default" | "discard";
}
