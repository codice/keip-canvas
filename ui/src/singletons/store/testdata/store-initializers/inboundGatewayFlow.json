{
  "nodes": [
    {
      "id": "_k0Z45PtEc",
      "type": "eipNode",
      "position": { "x": 0, "y": 0 },
      "data": { "followerId": "hF-dVMmHqZ", "label": "testIn" },
      "measured": { "width": 109, "height": 116 },
      "selected": false,
      "dragging": false,
      "targetPosition": "left",
      "sourcePosition": "right"
    },
    {
      "id": "hF-dVMmHqZ",
      "type": "followerNode",
      "position": { "x": 358, "y": 12 },
      "data": { "leaderId": "_k0Z45PtEc" },
      "measured": { "width": 114, "height": 92 },
      "selected": false,
      "dragging": false,
      "targetPosition": "left",
      "sourcePosition": "right"
    },
    {
      "id": "QvmTf6Jm4O",
      "type": "eipNode",
      "position": { "x": 184, "y": 13 },
      "data": {},
      "measured": { "width": 99, "height": 90 },
      "selected": false,
      "dragging": false,
      "targetPosition": "left",
      "sourcePosition": "right"
    }
  ],
  "edges": [
    {
      "source": "_k0Z45PtEc",
      "sourceHandle": "output",
      "target": "QvmTf6Jm4O",
      "targetHandle": "input",
      "id": "xy-edge___k0Z45PtEcoutput-QvmTf6Jm4Oinput"
    },
    {
      "source": "QvmTf6Jm4O",
      "sourceHandle": "output",
      "target": "hF-dVMmHqZ",
      "targetHandle": "input",
      "id": "xy-edge__QvmTf6Jm4Ooutput-hF-dVMmHqZinput"
    }
  ],
  "eipConfigs": {
    "_k0Z45PtEc": {
      "attributes": {},
      "children": [],
      "eipId": { "namespace": "http", "name": "inbound-gateway" }
    },
    "hF-dVMmHqZ": {
      "attributes": {},
      "children": [],
      "eipId": { "namespace": "integration", "name": "channel" }
    },
    "QvmTf6Jm4O": {
      "attributes": {},
      "children": [],
      "eipId": { "namespace": "integration", "name": "transformer" }
    }
  },
  "version": "1.0"
}
