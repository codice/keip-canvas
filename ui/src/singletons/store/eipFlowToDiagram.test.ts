import { beforeEach, describe, expect, test, vi } from "vitest"
import { EipFlow } from "../../api/generated/eipFlow"
import { eipFlowToDiagram } from "./eipFlowToDiagram"
import customEntityFlow from "./testdata/eipFlows/customEntity.json"
import inboundGatewayFlow from "./testdata/eipFlows/inboundGateway.json"
import nestedChildrenFlow from "./testdata/eipFlows/nestedChildren.json"
import payloadTypeRouterFlow from "./testdata/eipFlows/routers/payloadTypeRouter.json"
import recipientListRouterFlow from "./testdata/eipFlows/routers/recipientListRouter.json"
import simpleRouterFlow from "./testdata/eipFlows/routers/simpleRouter.json"
import xpathRouterFlow from "./testdata/eipFlows/routers/xpathRouter.json"
import simpleFlow from "./testdata/eipFlows/simple.json"

vi.mock("zustand")

let nodeCount = 0
let childCount = 0

// Generate predictable node ids
vi.mock("../../utils/nodeIdGenerator", () => ({
  generateChildId: () => `c${childCount++}`,
  generateNodeId: () => `n${nodeCount++}`,
}))

beforeEach(() => {
  nodeCount = 0
  childCount = 0
})

describe("EipFlow to diagram", () => {
  test.each([
    {
      msg: "simple flow",
      flow: simpleFlow,
    },
    {
      msg: "nested children",
      flow: nestedChildrenFlow,
    },
    {
      msg: "simple router",
      flow: simpleRouterFlow,
    },
    {
      msg: "payload-type router",
      flow: payloadTypeRouterFlow,
    },
    {
      msg: "recipient-list router",
      flow: recipientListRouterFlow,
    },
    {
      msg: "xpath router",
      flow: xpathRouterFlow,
    },
    {
      msg: "inbound gateway",
      flow: inboundGatewayFlow,
    },
    {
      msg: "custom entity",
      flow: customEntityFlow,
    },
  ])("$msg", ({ flow }) => {
    const flowDiagram = eipFlowToDiagram(flow as unknown as EipFlow)
    expect(flowDiagram).toMatchSnapshot()
  })
})
