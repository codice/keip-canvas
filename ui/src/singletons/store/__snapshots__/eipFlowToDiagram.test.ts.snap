// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EipFlow to diagram > 'custom entity' 1`] = `
{
  "customEntities": {
    "e1": "<bean class="com.example.Test"><property name="limit" value="65536"/></bean>",
  },
  "edges": [],
  "eipConfigs": {},
  "nodes": [],
}
`;

exports[`EipFlow to diagram > 'inbound gateway' 1`] = `
{
  "customEntities": {},
  "edges": [
    {
      "id": "ch-gatewayIn-updatePayload",
      "source": "n0",
      "target": "n1",
    },
    {
      "id": "ch-updatePayload-gatewayIn-reply-channel",
      "source": "n1",
      "target": "n2",
    },
  ],
  "eipConfigs": {
    "n0": {
      "attributes": {},
      "children": [],
      "eipId": {
        "name": "inbound-gateway",
        "namespace": "http",
      },
    },
    "n1": {
      "attributes": {},
      "children": [],
      "eipId": {
        "name": "transformer",
        "namespace": "integration",
      },
    },
    "n2": {
      "attributes": {},
      "children": [],
      "eipId": {
        "name": "channel",
        "namespace": "integration",
      },
    },
  },
  "nodes": [
    {
      "data": {
        "followerId": "n2",
        "label": "gatewayIn",
      },
      "id": "n0",
      "position": {
        "x": 0,
        "y": 55.75,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "updatePayload",
      },
      "id": "n1",
      "position": {
        "x": 203,
        "y": 0,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "leaderId": "n0",
      },
      "id": "n2",
      "position": {
        "x": 406,
        "y": 55.75,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "followerNode",
    },
  ],
}
`;

exports[`EipFlow to diagram > 'nested children' 1`] = `
{
  "customEntities": {},
  "edges": [
    {
      "id": "ch-in1-addPrefix",
      "source": "n0",
      "target": "n1",
    },
    {
      "id": "ch-addPrefix-out1",
      "source": "n1",
      "target": "n2",
    },
  ],
  "eipConfigs": {
    "c0": {
      "attributes": {
        "fixed-rate": "3000",
      },
      "children": [],
      "eipId": {
        "name": "poller",
        "namespace": "integration",
      },
    },
    "c1": {
      "attributes": {
        "maximum": "30000",
      },
      "children": [],
      "eipId": {
        "name": "exponential-back-off",
        "namespace": "integration",
      },
    },
    "c2": {
      "attributes": {},
      "children": [
        "c1",
      ],
      "eipId": {
        "name": "retry-advice",
        "namespace": "integration",
      },
    },
    "c3": {
      "attributes": {},
      "children": [
        "c2",
      ],
      "eipId": {
        "name": "request-handler-advice-chain",
        "namespace": "integration",
      },
    },
    "n0": {
      "attributes": {
        "expression": "'abc'",
      },
      "children": [
        "c0",
      ],
      "eipId": {
        "name": "inbound-channel-adapter",
        "namespace": "integration",
      },
    },
    "n1": {
      "attributes": {
        "expression": "'test_' + payload",
      },
      "children": [
        "c3",
      ],
      "eipId": {
        "name": "transformer",
        "namespace": "integration",
      },
    },
    "n2": {
      "attributes": {
        "destination-name": "test_queue",
        "pub-sub-domain": "false",
      },
      "children": [],
      "eipId": {
        "name": "outbound-channel-adapter",
        "namespace": "jms",
      },
    },
  },
  "nodes": [
    {
      "data": {
        "label": "in1",
      },
      "id": "n0",
      "position": {
        "x": 0,
        "y": 0,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "addPrefix",
      },
      "id": "n1",
      "position": {
        "x": 203,
        "y": 0,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "out1",
      },
      "id": "n2",
      "position": {
        "x": 406,
        "y": 0,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
  ],
}
`;

exports[`EipFlow to diagram > 'payload-type router' 1`] = `
{
  "customEntities": {},
  "edges": [
    {
      "id": "ch-in1-router",
      "source": "n0",
      "target": "n1",
    },
    {
      "animated": true,
      "data": {
        "mapping": {
          "mapperId": {
            "name": "mapping",
            "namespace": "integration",
          },
          "matcher": {
            "description": "The type of the payload that this mapping will match, e.g. 'java.lang.String' or 'java.lang.Integer' ",
            "name": "type",
            "required": false,
            "type": "string",
          },
          "matcherValue": "Integer",
        },
      },
      "id": "ch-router-out2",
      "source": "n1",
      "target": "n2",
      "type": "dynamicEdge",
    },
    {
      "animated": true,
      "data": {
        "mapping": {
          "mapperId": {
            "name": "mapping",
            "namespace": "integration",
          },
          "matcher": {
            "description": "The type of the payload that this mapping will match, e.g. 'java.lang.String' or 'java.lang.Integer' ",
            "name": "type",
            "required": false,
            "type": "string",
          },
          "matcherValue": "Boolean",
        },
      },
      "id": "ch-router-logger",
      "source": "n1",
      "target": "n3",
      "type": "dynamicEdge",
    },
    {
      "animated": true,
      "data": {
        "mapping": {
          "mapperId": {
            "name": "mapping",
            "namespace": "integration",
          },
          "matcher": {
            "description": "The type of the payload that this mapping will match, e.g. 'java.lang.String' or 'java.lang.Integer' ",
            "name": "type",
            "required": false,
            "type": "string",
          },
          "matcherValue": "String",
        },
      },
      "id": "ch-router-out1",
      "source": "n1",
      "target": "n4",
      "type": "dynamicEdge",
    },
  ],
  "eipConfigs": {
    "c0": {
      "attributes": {
        "channel": "ch-router-out1",
        "type": "String",
      },
      "children": [],
      "eipId": {
        "name": "mapping",
        "namespace": "integration",
      },
    },
    "c1": {
      "attributes": {
        "channel": "ch-router-out2",
        "type": "Integer",
      },
      "children": [],
      "eipId": {
        "name": "mapping",
        "namespace": "integration",
      },
    },
    "c2": {
      "attributes": {
        "channel": "ch-router-logger",
        "type": "Boolean",
      },
      "children": [],
      "eipId": {
        "name": "mapping",
        "namespace": "integration",
      },
    },
    "n0": {
      "attributes": {
        "expression": "'abc'",
      },
      "children": [],
      "eipId": {
        "name": "inbound-channel-adapter",
        "namespace": "integration",
      },
    },
    "n1": {
      "attributes": {},
      "children": [],
      "eipId": {
        "name": "payload-type-router",
        "namespace": "integration",
      },
    },
    "n2": {
      "attributes": {},
      "children": [],
      "eipId": {
        "name": "outbound-channel-adapter",
        "namespace": "http",
      },
    },
    "n3": {
      "attributes": {},
      "children": [],
      "eipId": {
        "name": "logging-channel-adapter",
        "namespace": "integration",
      },
    },
    "n4": {
      "attributes": {
        "destination-name": "test_queue",
        "pub-sub-domain": "false",
      },
      "children": [],
      "eipId": {
        "name": "outbound-channel-adapter",
        "namespace": "jms",
      },
    },
  },
  "nodes": [
    {
      "data": {
        "label": "in1",
      },
      "id": "n0",
      "position": {
        "x": 0,
        "y": 203,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "router",
      },
      "id": "n1",
      "position": {
        "x": 203,
        "y": 203,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "out2",
      },
      "id": "n2",
      "position": {
        "x": 406,
        "y": 0,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "logger",
      },
      "id": "n3",
      "position": {
        "x": 406,
        "y": 203,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "out1",
      },
      "id": "n4",
      "position": {
        "x": 406,
        "y": 406,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
  ],
}
`;

exports[`EipFlow to diagram > 'recipient-list router' 1`] = `
{
  "customEntities": {},
  "edges": [
    {
      "id": "ch-in1-router",
      "source": "n0",
      "target": "n1",
    },
    {
      "animated": true,
      "data": {
        "mapping": {
          "mapperId": {
            "name": "recipient",
            "namespace": "integration",
          },
          "matcher": {
            "description": "An expression to be evaluated to determine if this recipient should be included in the recipient list for a given input Message. The evaluation result of the expression must be a boolean. If this attribute is not defined, the channel will always be among the list of recipients.",
            "name": "selector-expression",
            "required": false,
            "type": "string",
          },
        },
      },
      "id": "ch-router-out2",
      "source": "n1",
      "target": "n2",
      "type": "dynamicEdge",
    },
    {
      "animated": true,
      "data": {
        "mapping": {
          "mapperId": {
            "name": "recipient",
            "namespace": "integration",
          },
          "matcher": {
            "description": "An expression to be evaluated to determine if this recipient should be included in the recipient list for a given input Message. The evaluation result of the expression must be a boolean. If this attribute is not defined, the channel will always be among the list of recipients.",
            "name": "selector-expression",
            "required": false,
            "type": "string",
          },
          "matcherValue": "headers['log'] == true",
        },
      },
      "id": "ch-router-logger",
      "source": "n1",
      "target": "n3",
      "type": "dynamicEdge",
    },
    {
      "animated": true,
      "data": {
        "mapping": {
          "mapperId": {
            "name": "recipient",
            "namespace": "integration",
          },
          "matcher": {
            "description": "An expression to be evaluated to determine if this recipient should be included in the recipient list for a given input Message. The evaluation result of the expression must be a boolean. If this attribute is not defined, the channel will always be among the list of recipients.",
            "name": "selector-expression",
            "required": false,
            "type": "string",
          },
        },
      },
      "id": "ch-router-out1",
      "source": "n1",
      "target": "n4",
      "type": "dynamicEdge",
    },
  ],
  "eipConfigs": {
    "c0": {
      "attributes": {
        "channel": "ch-router-out1",
      },
      "children": [],
      "eipId": {
        "name": "recipient",
        "namespace": "integration",
      },
    },
    "c1": {
      "attributes": {
        "channel": "ch-router-out2",
      },
      "children": [],
      "eipId": {
        "name": "recipient",
        "namespace": "integration",
      },
    },
    "c2": {
      "attributes": {
        "channel": "ch-router-logger",
        "selector-expression": "headers['log'] == true",
      },
      "children": [],
      "eipId": {
        "name": "recipient",
        "namespace": "integration",
      },
    },
    "n0": {
      "attributes": {
        "expression": "'abc'",
      },
      "children": [],
      "eipId": {
        "name": "inbound-channel-adapter",
        "namespace": "integration",
      },
    },
    "n1": {
      "attributes": {
        "apply-sequence": "true",
      },
      "children": [],
      "eipId": {
        "name": "recipient-list-router",
        "namespace": "integration",
      },
    },
    "n2": {
      "attributes": {},
      "children": [],
      "eipId": {
        "name": "outbound-channel-adapter",
        "namespace": "http",
      },
    },
    "n3": {
      "attributes": {},
      "children": [],
      "eipId": {
        "name": "logging-channel-adapter",
        "namespace": "integration",
      },
    },
    "n4": {
      "attributes": {
        "destination-name": "test_queue",
        "pub-sub-domain": "false",
      },
      "children": [],
      "eipId": {
        "name": "outbound-channel-adapter",
        "namespace": "jms",
      },
    },
  },
  "nodes": [
    {
      "data": {
        "label": "in1",
      },
      "id": "n0",
      "position": {
        "x": 0,
        "y": 203,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "router",
      },
      "id": "n1",
      "position": {
        "x": 203,
        "y": 203,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "out2",
      },
      "id": "n2",
      "position": {
        "x": 406,
        "y": 0,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "logger",
      },
      "id": "n3",
      "position": {
        "x": 406,
        "y": 203,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "out1",
      },
      "id": "n4",
      "position": {
        "x": 406,
        "y": 406,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
  ],
}
`;

exports[`EipFlow to diagram > 'simple flow' 1`] = `
{
  "customEntities": {},
  "edges": [
    {
      "id": "ch-in1-addPrefix",
      "source": "n0",
      "target": "n1",
    },
    {
      "id": "ch-addPrefix-out1",
      "source": "n1",
      "target": "n2",
    },
  ],
  "eipConfigs": {
    "n0": {
      "attributes": {
        "expression": "'abc'",
      },
      "children": [],
      "eipId": {
        "name": "inbound-channel-adapter",
        "namespace": "integration",
      },
    },
    "n1": {
      "attributes": {
        "expression": "'test_' + payload",
      },
      "children": [],
      "eipId": {
        "name": "transformer",
        "namespace": "integration",
      },
    },
    "n2": {
      "attributes": {
        "destination-name": "test_queue",
        "pub-sub-domain": "false",
      },
      "children": [],
      "eipId": {
        "name": "outbound-channel-adapter",
        "namespace": "jms",
      },
    },
  },
  "nodes": [
    {
      "data": {
        "label": "in1",
      },
      "id": "n0",
      "position": {
        "x": 0,
        "y": 0,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "addPrefix",
      },
      "id": "n1",
      "position": {
        "x": 203,
        "y": 0,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "out1",
      },
      "id": "n2",
      "position": {
        "x": 406,
        "y": 0,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
  ],
}
`;

exports[`EipFlow to diagram > 'simple router' 1`] = `
{
  "customEntities": {},
  "edges": [
    {
      "id": "ch-in1-testRouter",
      "source": "n0",
      "target": "n1",
    },
    {
      "animated": true,
      "data": {
        "mapping": {
          "mapperId": {
            "name": "mapping",
            "namespace": "integration",
          },
          "matcher": {
            "description": "A value of the evaluation token that will be mapped to a channel reference (e.g., mapping value='foo' channel='myChannel')",
            "name": "value",
            "required": false,
            "type": "string",
          },
          "matcherValue": "first",
        },
      },
      "id": "ch-testRouter-out1",
      "source": "n1",
      "target": "n2",
      "type": "dynamicEdge",
    },
    {
      "animated": true,
      "data": {
        "mapping": {
          "mapperId": {
            "name": "mapping",
            "namespace": "integration",
          },
          "matcher": {
            "description": "A value of the evaluation token that will be mapped to a channel reference (e.g., mapping value='foo' channel='myChannel')",
            "name": "value",
            "required": false,
            "type": "string",
          },
          "matcherValue": "second",
        },
      },
      "id": "ch-testRouter-out2",
      "source": "n1",
      "target": "n3",
      "type": "dynamicEdge",
    },
    {
      "animated": true,
      "data": {
        "mapping": {
          "isDefaultMapping": true,
          "mapperId": {
            "name": "mapping",
            "namespace": "integration",
          },
          "matcher": {
            "description": "A value of the evaluation token that will be mapped to a channel reference (e.g., mapping value='foo' channel='myChannel')",
            "name": "value",
            "required": false,
            "type": "string",
          },
        },
      },
      "id": "ch-testRouter-logger",
      "source": "n1",
      "target": "n4",
      "type": "dynamicEdge",
    },
  ],
  "eipConfigs": {
    "c0": {
      "attributes": {
        "channel": "ch-testRouter-out1",
        "value": "first",
      },
      "children": [],
      "eipId": {
        "name": "mapping",
        "namespace": "integration",
      },
    },
    "c1": {
      "attributes": {
        "channel": "ch-testRouter-out2",
        "value": "second",
      },
      "children": [],
      "eipId": {
        "name": "mapping",
        "namespace": "integration",
      },
    },
    "n0": {
      "attributes": {
        "expression": "'abc'",
      },
      "children": [],
      "eipId": {
        "name": "inbound-channel-adapter",
        "namespace": "integration",
      },
    },
    "n1": {
      "attributes": {
        "send-timeout": "2000",
      },
      "children": [],
      "eipId": {
        "name": "router",
        "namespace": "integration",
      },
      "routerKey": {
        "attributes": {
          "expression": "payload",
        },
        "eipId": {
          "name": "expression",
          "namespace": "integration",
        },
      },
    },
    "n2": {
      "attributes": {
        "destination-name": "test_queue",
        "pub-sub-domain": "false",
      },
      "children": [],
      "eipId": {
        "name": "outbound-channel-adapter",
        "namespace": "jms",
      },
    },
    "n3": {
      "attributes": {},
      "children": [],
      "eipId": {
        "name": "outbound-channel-adapter",
        "namespace": "http",
      },
    },
    "n4": {
      "attributes": {},
      "children": [],
      "eipId": {
        "name": "logging-channel-adapter",
        "namespace": "integration",
      },
    },
  },
  "nodes": [
    {
      "data": {
        "label": "in1",
      },
      "id": "n0",
      "position": {
        "x": 0,
        "y": 203,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "testRouter",
      },
      "id": "n1",
      "position": {
        "x": 203,
        "y": 203,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "out1",
      },
      "id": "n2",
      "position": {
        "x": 406,
        "y": 0,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "out2",
      },
      "id": "n3",
      "position": {
        "x": 406,
        "y": 203,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "logger",
      },
      "id": "n4",
      "position": {
        "x": 406,
        "y": 406,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
  ],
}
`;

exports[`EipFlow to diagram > 'xpath router' 1`] = `
{
  "customEntities": {},
  "edges": [
    {
      "id": "ch-in1-router",
      "source": "n0",
      "target": "n1",
    },
    {
      "animated": true,
      "data": {
        "mapping": {
          "mapperId": {
            "name": "mapping",
            "namespace": "int-xml",
          },
          "matcher": {
            "description": "A value of the evaluation token that will be mapped to a channel reference (e.g., mapping value='foo' channel='myChannel')",
            "name": "value",
            "required": false,
            "type": "string",
          },
          "matcherValue": "second",
        },
      },
      "id": "ch-router-out2",
      "source": "n1",
      "target": "n2",
      "type": "dynamicEdge",
    },
    {
      "animated": true,
      "data": {
        "mapping": {
          "mapperId": {
            "name": "mapping",
            "namespace": "int-xml",
          },
          "matcher": {
            "description": "A value of the evaluation token that will be mapped to a channel reference (e.g., mapping value='foo' channel='myChannel')",
            "name": "value",
            "required": false,
            "type": "string",
          },
          "matcherValue": "third",
        },
      },
      "id": "ch-router-logger",
      "source": "n1",
      "target": "n3",
      "type": "dynamicEdge",
    },
    {
      "animated": true,
      "data": {
        "mapping": {
          "mapperId": {
            "name": "mapping",
            "namespace": "int-xml",
          },
          "matcher": {
            "description": "A value of the evaluation token that will be mapped to a channel reference (e.g., mapping value='foo' channel='myChannel')",
            "name": "value",
            "required": false,
            "type": "string",
          },
          "matcherValue": "first",
        },
      },
      "id": "ch-router-out1",
      "source": "n1",
      "target": "n4",
      "type": "dynamicEdge",
    },
  ],
  "eipConfigs": {
    "c1": {
      "attributes": {
        "channel": "ch-router-out1",
        "value": "first",
      },
      "children": [],
      "eipId": {
        "name": "mapping",
        "namespace": "int-xml",
      },
    },
    "c2": {
      "attributes": {
        "channel": "ch-router-out2",
        "value": "second",
      },
      "children": [],
      "eipId": {
        "name": "mapping",
        "namespace": "int-xml",
      },
    },
    "c3": {
      "attributes": {
        "channel": "ch-router-logger",
        "value": "third",
      },
      "children": [],
      "eipId": {
        "name": "mapping",
        "namespace": "int-xml",
      },
    },
    "n0": {
      "attributes": {
        "expression": "'abc'",
      },
      "children": [],
      "eipId": {
        "name": "inbound-channel-adapter",
        "namespace": "integration",
      },
    },
    "n1": {
      "attributes": {
        "phase": "init",
      },
      "children": [],
      "eipId": {
        "name": "xpath-router",
        "namespace": "int-xml",
      },
      "routerKey": {
        "attributes": {
          "expression": "/test/xpath",
          "id": "token",
          "ns-prefix": "testing",
          "ns-uri": "test.example.com",
        },
        "eipId": {
          "name": "xpath-expression",
          "namespace": "int-xml",
        },
      },
    },
    "n2": {
      "attributes": {},
      "children": [],
      "eipId": {
        "name": "outbound-channel-adapter",
        "namespace": "http",
      },
    },
    "n3": {
      "attributes": {},
      "children": [],
      "eipId": {
        "name": "logging-channel-adapter",
        "namespace": "integration",
      },
    },
    "n4": {
      "attributes": {
        "destination-name": "test_queue",
        "pub-sub-domain": "false",
      },
      "children": [],
      "eipId": {
        "name": "outbound-channel-adapter",
        "namespace": "jms",
      },
    },
  },
  "nodes": [
    {
      "data": {
        "label": "in1",
      },
      "id": "n0",
      "position": {
        "x": 0,
        "y": 203,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "router",
      },
      "id": "n1",
      "position": {
        "x": 203,
        "y": 203,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "out2",
      },
      "id": "n2",
      "position": {
        "x": 406,
        "y": 0,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "logger",
      },
      "id": "n3",
      "position": {
        "x": 406,
        "y": 203,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
    {
      "data": {
        "label": "out1",
      },
      "id": "n4",
      "position": {
        "x": 406,
        "y": 406,
      },
      "sourcePosition": "right",
      "targetPosition": "left",
      "type": "eipNode",
    },
  ],
}
`;
